pkg/auth/oauth/handlers/interfaces.go

```diff
@@ -25,6 +25,51 @@ type AuthenticationRedirector interface {
 	AuthenticationRedirect(w http.ResponseWriter, req *http.Request) (err error)
 }
 
+// AuthenticationRedirectors is a collection that provides a list of
+// acceptable authentication mechanisms
+type AuthenticationRedirectors struct {
+	names         []string
+	redirectorMap map[string]AuthenticationRedirector
+}
+
+// Add a name and a matching redirection routine to the list
+// The entries added here will be retained in FIFO ordering.
+func (ar *AuthenticationRedirectors) Add(name string, redirector AuthenticationRedirector) {
+	// Initialize the map the first time
+	if ar.redirectorMap == nil {
+		ar.redirectorMap = make(map[string]AuthenticationRedirector, 1)
+	}
+
+	// If this name already exists in the map, ignore it.
+	// This should be impossible, as uniqueness is tested by the config
+	// validator, but it's always best to catch mistakes.
+	if ar.redirectorMap[name] != nil {
```

Comment:
Ah, right. I had to use that construct when it was a loop, but I can do it the right way now that it's an "if"